Makefile

```diff
@@ -202,22 +207,12 @@ $(SECURITY_GO_BINS): depend
 .PHONY: go-build
 go-build: $(PILOT_GO_BINS) $(MIXER_GO_BINS) $(SECURITY_GO_BINS)
 
-# The following are convenience aliases for most of the go targets
-# The first block is for aliases that are the same as the actual binary,
-# while the ones that follow need slight adjustments to their names.
-
-IDENTITY_ALIAS_LIST:=istioctl mixc mixs pilot-agent servicegraph sidecar-initializer
-.PHONY: $(IDENTITY_ALIAS_LIST)
-$(foreach ITEM,$(IDENTITY_ALIAS_LIST),$(eval $(ITEM): ${ISTIO_BIN}/$(ITEM)))
-
-.PHONY: istio-ca
-istio-ca: ${ISTIO_BIN}/istio_ca
-
-.PHONY: node-agent
-node-agent: ${ISTIO_BIN}/node_agent
+MAPPING_MODULES=pilot.pilot-discovery proxy.pilot-agent istioctl.istioctl sidecar_initializer.sidecar-initializer mixs.mixs mixc.mixc servicegraph.servicegraph node-agent.node_agent istio-ca.istio_ca
+MODULES=pilot proxy istioctl sidecar_initializer mixs mixc servicegraph node-agent istio-ca
 
-.PHONY: pilot
-pilot: ${ISTIO_BIN}/pilot-discovery
+# The following are convenience aliases for the Go targets built without debugger information
+.PHONY: $(foreach MOD,${MODULES},${MOD})
```

Comment:
I think this can be expressed as:
.PHONY: $(MODULES)

another approach is to try something like:

$(foreach MOD, $(MAPPING_MODULES), $(eval .PHONY: $(basename $(MOD) ))

and then there's no need for the MODULES list.  A possible further step is to combine the loops together:

<pre>
$(foreach MAP_MOD,${MAPPING_MODULES}, \
    $(eval .PHONY: $(basename $(MAP_MOD) )  \
    $(eval $(basename ${MAP_MOD}): ${ISTIO_BIN}/$(subst .,,$(suffix ${MAP_MOD}))))
</pre>

It should all work together on the same line but I split it across 3 for readability.

Reply:
looks like a good idea